grails-app/services/org/pih/warehouse/core/OrganizationService.groovy

```diff
@@ -67,6 +67,13 @@ class OrganizationService {
         return organization
     }
 
+    Organization saveOrganization(Organization organization) {
+        if (organization.validate() && !organization.hasErrors()) {
```

Comment:
@alannadolny the order should be reverted, so:
```groovy
!organization.hasErrors() && organization.validate()
```
@jmiranda @alannadolny in this case the `hasErrors()` might have errors even before the `validate()` is executed.
It is because when you are **binding** data, e.g. `new Object(params)` or like it is done here in the controller using the `bindData`, then a validation of **data types** is proceeded - so the `organization.errors` might not be empty at this point if e.g. we provided a `name` that would be an `Integer` - it would already contain that error.
`validate()` on the other hand proceeds with the **constraints** validation, so to check for example if the name is unique, etc.

So I'd say it might be a good pattern to first check `hasErrors()`, not to invoke the `validate()` (which is "deeper" than when binding data, that only checks the data type), if there are already errors from data types.
On the other hand it might be good to let the user know all the errors immediately, so that he/she can correct him/herself once.

I'm fine with both, just wanted to point out, that a different validation is proceeded when binding data and when calling `validate()`.